model/task_lifecycle.go

```diff
@@ -612,11 +612,37 @@ func MarkTaskDispatched(t *task.Task, hostId, distroId string) error {
 // RestartFailedTasks attempts to restart failed tasks that started between 2 times
 // It returns a slice of task IDs that were successfully restarted as well as a slice
 // of task IDs that failed to restart
-func RestartFailedTasks(startTime, endTime time.Time, user string, dryRun bool) ([]string, []string, error) {
+// the dryRun parameter will return the tasks that will be restarted if sent true
+// the red and purple params will only restart tasks that were failed due to the test
+// or due to the system, respectively
+func RestartFailedTasks(startTime, endTime time.Time, user string,
+	dryRun, onlyRed, onlyPurple bool) ([]string, []string, error) {
```

Comment:
can we have a struct for options and a results struct to make the API for this function more reasonable?

Reply:
done